Block/Transparent/Redirect.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2020 Adyen BV (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Block\Transparent;
+
+use Adyen\Service\Validator\DataArrayValidator;
+use Magento\Framework\View\Element\Template;
+use Magento\Framework\UrlInterface;
+
+class Redirect extends Template
+{
+    /**
+     * @var UrlInterface
+     */
+    private $url;
+    /**
+     * @var \Adyen\Payment\Logger\AdyenLogger
+     */
+    protected $adyenLogger;
+    /**
+     * Redirect constructor.
+     * @param Template\Context $context
+     * @param \Magento\Framework\Url $url
+     * @param array $data
+     */
+    public function __construct(
+        Template\Context $context,
+        \Magento\Framework\Url $url,
+        \Adyen\Payment\Logger\AdyenLogger $adyenLogger,
+        array $data = []
+    ) {
+        $this->url = $url;
+        $this->adyenLogger = $adyenLogger;
+        parent::__construct($context, $data);
+    }
+
+    /**
+     * Returns url for redirect.
+     * @return string|null
+     */
+    public function getRedirectUrl()
+    {
+        return $this->url->getUrl("adyen/process/redirect");
```

Comment:
We should test and use Adyen\Payment\Helper\Data::getOrigin() here to build this redirect endpoint's URL also for the PWA integrations that might use it

Reply:
We created a separate ticket to investigate it

Reply:
Ok let's add a comment to inform everyone about this being replaced soon
```suggestion
        return $this->url->getUrl("adyen/process/redirect"); //TODO this will be replaced by getOrigin() for PWA integrations
```